core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1476,6 +1476,11 @@ class Log(@volatile var dir: File,
         if (fetchInfo == null) {
           segmentEntry = segments.higherEntry(segmentEntry.getKey)
         } else {
+          // record consumer lag metrics
+          if (isolation != FetchLogEnd) {
```

Comment:
We could pass in a boolean `recordLagMetrics` or something like this instead of implying this based on the isolation level.

Reply:
Yes, had thought about this, I was avoiding the signature bloating, but with your feedback I think it will make this more reliable and maintainable and less error prone to future isolation level behavior. I will make the change